azkaban-web-server/src/main/java/azkaban/imagemgmt/servlets/HPFlowServlet.java

```diff
@@ -0,0 +1,129 @@
+    this.hpFlowService = server.getHPFlowService();
+  }
+
+  @Override
+  protected void handleGet(final HttpServletRequest req,
+      final HttpServletResponse resp, final Session session)
+    throws ServletException, IOException {
+
+  }
+
+  @Override
+  protected void handlePost(final HttpServletRequest req,
+      final HttpServletResponse resp, final Session session)
+      throws ServletException, IOException {
+    handleAddHPFlows(req, resp, session);
+  }
+
+  /**
+   * This method takes in a CSV of high priority flows and adds them to metadata.
+   * @param req
+   * @param resp
+   * @param session
+   * @throws ServletException
+   * @throws IOException
+   */
+  protected void handleAddHPFlows(final HttpServletRequest req,
+      final HttpServletResponse resp, final Session session)
+      throws ServletException, IOException {
+    try {
+      final String jsonPayLoad = HttpRequestUtils.getBody(req);
+      final HPFlowDTO hpFlowDTO = this.converterUtils.convertToDTO(jsonPayLoad, HPFlowDTO.class);
+      // If there are no flow IDs in the list return with error
+      if (hpFlowDTO.getFlowIds().isEmpty()) {
+        log.error("There are no flow IDs provided");
+        throw new ImageMgmtValidationException(ErrorCode.BAD_REQUEST,
+            "Required field flow IDs is empty");
+      }
+      // Check for required permission to invoke the API
+      final User user = session.getUser();
+      if (!hasHPFlowManagementPermission(user)) {
+        log.debug(String.format("Invalid permission to access High Priority "
+            + "flows for user: %s", user.getUserId()));
+        throw new ImageMgmtInvalidPermissionException(ErrorCode.FORBIDDEN,
+            "Invalid permission to manage high priority flows");
+      }
+      // Set the user who invoked the API
+      hpFlowDTO.setCreatedBy(user.getUserId());
+      // Add the HP flow metadata
+      this.hpFlowService.addHPFlows(hpFlowDTO);
+      // Prepare to send response
+      resp.setStatus(HttpStatus.SC_CREATED);
+      sendResponse(resp, HttpServletResponse.SC_CREATED, new HashMap<>());
+    } catch (final ImageMgmtException e) {
+      log.error("Exception while adding high priority flows.", e);
+      sendErrorResponse(resp, e.getErrorCode().getCode(), e.getMessage());
+    } catch (final Exception e) {
+      log.error("Exception while adding high priority flows", e);
+      sendErrorResponse(resp, HttpServletResponse.SC_INTERNAL_SERVER_ERROR,
+          "Exception while adding high priority flows. " + e.getMessage());
+    }
+  }
+
+  /**
+   * Checks if the provided user has access to manage high priority flows.
+   * @param user
+   * @return
+   */
+  private boolean hasHPFlowManagementPermission(final User user) {
+    /**
+     * Azkaban ADMIN role must have full permission to access image management APIs. Hence, no
+     * further permission check is required.
+     */
+    if (isAzkabanAdmin(user)) {
+      return true;
+    }
```

Comment:
The new api endpoint adds maintenance overhead on the Azkaban team as it involves putting in place a process to add high priority flows. Why is this feature available to high priority flows only? What's a high priority flow?

Reply:
As the name suggests a flow which has high business value is a high priority flow. With containerization and image rampup feature, the jobtype owners want to make sure that a flow which has high priority runs on ACTIVE version rather than any version which is being ramped up and could be potentially unstable.

Reply:
The API will be eventually self served a list of owners.